MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs

```diff
@@ -0,0 +1,114 @@
+
+        /// <summary>
+        /// The <see cref="CompareInfo"/> instance.
+        /// </summary>
+        protected readonly CompareInfo _compareInfo = CultureInfo.InvariantCulture.CompareInfo;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaInfoResolver"/> class.
+        /// </summary>
+        /// <param name="localizationManager">The localization manager.</param>
+        /// <param name="mediaEncoder">The media encoder.</param>
+        /// <param name="namingOptions">The naming options.</param>
+        public MediaInfoResolver(
+            ILocalizationManager localizationManager,
+            IMediaEncoder mediaEncoder,
+            NamingOptions namingOptions)
+        {
+            _localizationManager = localizationManager;
+            _mediaEncoder = mediaEncoder;
+            _namingOptions = namingOptions;
+        }
+
+        /// <summary>
+        /// Detects the language of a specific media stream.
+        /// </summary>
+        /// <param name="mediaStream">The media stream.</param>
+        /// <param name="fileNameWithoutExtension">The filename without extension.</param>
+        /// <param name="videoFileNameWithoutExtension">The video filename without extension.</param>
+        protected MediaStream DetectLanguage(MediaStream mediaStream, string fileNameWithoutExtension, string videoFileNameWithoutExtension)
+        {
+            foreach (var separator in _namingOptions.MediaFlagDelimiters)
+            {
+                var languageString = fileNameWithoutExtension;
+
+                while (languageString.Length > videoFileNameWithoutExtension.Length)
+                {
+                    var lastDot = languageString.LastIndexOf(separator);
+
+                    if (lastDot < videoFileNameWithoutExtension.Length)
+                    {
+                        if (mediaStream.Title == null && languageString.Length != videoFileNameWithoutExtension.Length)
+                        {
+                            mediaStream.Title = languageString.Remove(0, videoFileNameWithoutExtension.Length + 1);
+                        }
+
+                        break;
+                    }
+
+                    string currentSlice = languageString[lastDot..];
+
+                    if (_namingOptions.MediaDefaultFlags.Any(s => currentSlice[1..].Contains(s, StringComparison.OrdinalIgnoreCase))
+                        || _namingOptions.MediaForcedFlags.Any(s => currentSlice[1..].Contains(s, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        languageString = languageString[..lastDot];
+                        continue;
+                    }
+
+                    // Try to translate to three character code
+                    var culture = _localizationManager.FindLanguageInfo(currentSlice[1..]);
+
+                    if (culture == null || mediaStream.Language != null)
```

Comment:
If the stream already has a language set (such as from an mks/mka container) should we check to see if the filename language matches and ignore it instead of saving it in the title?